### DISCUSSION TITLE
LXQt 1.1.0 will come soon (Done)

### DISCUSSION THREAD

**Discussion by tsujan** [lxqt/lxqt#2224] [General]
This is the "General" part of the announcement. Please tell me if an important highlight is missing or there are typos or grammatical mistakes. The detailed parts will be added after the release, based on the release notes.

 * LXQt 1.1.0 depends on Qt 5.15, the last LTS version of Qt5. (Porting to Qt6 was started, but it needed a stable version of KF6.)
 * LXQt's file manager now supports the file manager DBus interface, which is used by some apps (like Firefox and Chromium) that call the default file manager to show files inside folders or do other jobs. Also, the "Recent Files" item is added to the File menu. As usual, problems are fixed and functionalities are improved.
 * The new component `xdg-desktop-portal-lxqt` is added to LXQt, as a backend implementation for `xdg-desktop-portal`. When it is installed, some non-Qt apps like Firefox can be set up to use LXQt file dialog.
 * LXQt themes are improved. A new theme and several wallpapers are added. Also, matching Qt palettes are added for a fairly uniform look and feel with Qt widget styles like Fusion. They can be found and applied in LXQt Appearance Configuration → Widget Style → Qt Palette.
 * The bookmark functionality is improved a lot in QTerminal and several (old) issues in the drop-down mode are fixed.
 * In LXQt Penel, the legacy tray icons are now shown inside Status Notifier when the System Tray plugin is enabled. That removes the problem that the legacy System Tray had with auto-hiding panels.
 * Tray icons with the charge percentage inside them are added to LXQt Power Manager.
 * A simple option is added to LXQt Session Settings for the global screen scaling.
 * Translations have received many updates.
 * And other changes that can be found in change logs of LXQt components.

Tonight, I'll apply the prerelease patches to the latest master branches offline, to test all components. If no problem happens, I'll start releasing 1.1.0 tomorrow — hopefully.

@stefonarch, you could merge translation PRs today.

Please don't merge any other PR before the next release or tomorrow. I'll merge https://github.com/lxqt/xdg-desktop-portal-lxqt/pull/8 and make a prerelease PR for it *after releasing `libfm-qt`, tomorrow* — again, hopefully.

**Comment by stefonarch** [comment]
> The detailed parts will be added after the release, based on the release notes.
> 
You didn't understand me before I guess: they are all ready. 
The website article ordering works with timestamps - to have this one  here on top it has to be the last published.
 My idea is  to have them in this order on the main page:
1. This release post
2. Pcmanfm-qt release notes
3. QTerminal 
4. LXimage
5. maybe archiver or screengrab

All the other ones will end up automatically [here](https://lxqt-project.org/releases/).

Will prepare all this evening and make a PR, hopefully including  also a screenshot with 1.1.0

**Reply by tsujan** [reply]
> You didn't understand me before I guess: they are all ready.

I meant what I put https://github.com/lxqt/lxqt/releases, which will be added immediately after the release is finished.

The main site is yours. You could use parts of https://github.com/lxqt/lxqt/releases or do it as you see fit. Of course, the main site is the real place for the announcement, but I prefer the contents of https://github.com/lxqt/lxqt/releases to be orderly too, as before.

**Reply by tsujan** [reply]
In case I've caused a confusion, let me explain:

Anyone who releases a component should add a release note *as a part of the process*. So, I *must* prepare the release notes of all components beforehand; there's no other way. (I prepared them an hour ago.)

I just add some of them, which seem more important, to https://github.com/lxqt/lxqt/releases; nothing more.

EDIT: I corrected my explanation: Adding a release note is a part of the releasing process, not something that can be done after it.

**Reply by stefonarch** [reply]
Probably it's me adding confusion. Just a note: I didn't see any release note of https://github.com/lxqt/xdg-desktop-portal-lxqt 0.2.0, wasn't it due to be released together with 1.1.0? It's mentioned here.

**Reply by tsujan** [reply]
> Just a note: I didn't see any release note of https://github.com/lxqt/xdg-desktop-portal-lxqt 

I explained it at the end of my first comment. In short, I'll add it.

**Reply by stefonarch** [reply]
Missed that; I see.
Tell me if you like something to change/add here: https://github.com/lxqt/lxqt.github.io/pull/108/files#diff-c4e1764ff0f59fedb619ad1893d02e7c88c94c56dcc2fe29dd06e2a3ddb867e6

**Reply by tsujan** [reply]
@stefonarch 
I released `lxqt-build-tools`, `libfm-qt` and `pcmanfm-qt` and made a prerelease PR for `xdg-desktop-portal-lxqt`.

Will release all components after having a long sleep.

**Comment by tsujan** [comment]
The release is made: https://github.com/lxqt/lxqt/releases/tag/1.1.0

Please check all components and tell me if there's a problem. Of course, I've checked, compiled and installed them but, due to the repetitious nature of the job, I may have made mistakes.

**Comment by stefonarch** [comment]
I didnt encounter compiling problems and was not able to spot mistakes in the notes but that means little - my eyes are tired.
The website articles and screenshots are ready, just updated the main release article.
https://github.com/lxqt/lxqt.github.io/pull/108/files#

Tell me if the main release article is ok, so we can publish it. Here it's now  easteregg preparing.

**Reply by tsujan** [reply]
https://github.com/lxqt/lxqt.github.io/pull/108 seems OK to me. Its screenshots are nice too.

> ... but that means little - my eyes are tired.

That was my situation last night ;) After a long, deep sleep, my brain got organized.

**Comment by yan12125** [comment]
Testing 1.1.0, I noticed an issue with lxqt-themes https://github.com/lxqt/lxqt-themes/issues/69. I don't consider it severe as there is a simple workaround, so I pushed LXQt 1.1.0 to testing repos of Arch Linux :)

**Reply by tsujan** [reply]
@yan12125 
Thanks a lot!

As for that issue, I agree with you that it isn't severe. But if others think it is, a point release could be made.

**Reply by yan12125** [reply]
If you're going to release liblxqt 1.1.1 with https://github.com/lxqt/liblxqt/pull/308, please ping me, thanks!

**Reply by tsujan** [reply]
I thought it might be better to make a point release after the split `libqtxdg` and your PR for `lxqt-session` are merged. But if you think a point release of liblxqt is needed now, it could be made easily. Should I make it after merging https://github.com/lxqt/liblxqt/pull/308?

I should also take a look at the code of `lxqt-config` and see if it needs a change after https://github.com/lxqt/liblxqt/pull/308 (probably not).

**Reply by yan12125** [reply]
A point release for liblxqt is not urgent, and making it along with qtxdg split is OK. It's just a reminder for me as I may miss upcoming releases among so many GitHub notifications :)

**Reply by tsujan** [reply]
I understand.

Also, the following components will need point releases alongside `libqtxdg`, `liblxqt` and `lxqt-session`:

lxqt-config
lxqt-notificationd
lxqt-panel
lxqt-powermanagement
lxqt-qtplugin
lxqt-runner
screengrab

**Reply by yan12125** [reply]
Are there so many necessary changes?

**Reply by tsujan** [reply]
> Are there so many necessary changes?

They depend on `libqtxdg`. You're more familiar with these things. Shouldn't we make them depend on libqtxdg-3.9.1 and release them with it?

**Reply by yan12125** [reply]
Yeah those depend on libqtxdg, but the qtxdg split only moves command line tools. Libraries remain unchanged (assuming no other PRs are merged before 1.1.1), so I think new releases for lxqt-config, ... are not necessary?

**Reply by tsujan** [reply]
> ... are not necessary?

For example, won't `screengrab` be broken after libqtxdg 3.9.0 → 3.9.1?

**Reply by yan12125** [reply]
Hmm, what kind of breakage did you refer to?

There is a kind of potential breakage for patch version updates (+0.0.1) in libraries: when a program/library uses private classes/functions in the updated library, and some of private stuffs are changed in the latest version. For example, libqtxdg defines private stuffs in `xdgiconloader_p.h`, so lxqt-qtplugin _might_ be broken as it uses private stuffs from libqtxdg. For such breakage, Arch Linux packagers rebuild relevant packages with the same upstream version and bump the distro-specific version (`pkgrel` for Arch). Anyway, https://github.com/lxqt/libqtxdg/blob/master/src/xdgiconloader/xdgiconloader_p.h is not changed since libqtxdg 3.7.0.

By the way, I tested screengrab 2.4.0 (which is built with libqtxdg 3.9.0) with libqtxdg 3.8.0 and it works fine. I don't think it will break with libqtxdg 3.9.1, either.

**Reply by yan12125** [reply]
@tsujan: oops, forgot to ask another question: do you plan to release liblxqt 1.1.1 soon (ex: this weekend) or do you prefer to release them along with other components after Qt 6 preparation PRs are merged? Either way is fine. I'm asking because I'd like to move LXQt 1.1 on Arch to stable repos. If liblxqt 1.1.1 is coming soon, I will wait and move LXQt after liblxqt is updated. Otherwise, I will backport https://github.com/lxqt/liblxqt/pull/308 to liblxqt 1.1.0 on Arch and then move LXQt.

**Reply by tsujan** [reply]
> I don't think it will break with libqtxdg 3.9.1, either.

OK. So, there's no need to the list I mentioned above. Probably, I'd seen another kind of problem before.

> do you plan to release liblxqt 1.1.1 soon (ex: this weekend) or do you prefer to release them along with other components after Qt 6 preparation PRs are merged?

The latter (with other components after Qt6-related PRs are merged).

**Reply by yan12125** [reply]
> The latter (with other components after Qt6-related PRs are merged).

No problem! I pushed liblxqt 1.1.0-2 to Arch testing. Will move LXQt 1.1 to Arch stable by the end of this weekend if there are no other reports.

**Reply by tsujan** [reply]
How did it become  1.1.0-2 and not 1.1.0-1?

**Reply by tsujan** [reply]
So, after merging the changes related to Qt6, only these components should be released: `libqtxdg`, `qtxdg-tools` (new), `lxqt-session`, `liblxqt` and `lxqt-config` (the last two only because of theme name case-insensitivity: https://github.com/lxqt/liblxqt/pull/308 and https://github.com/lxqt/lxqt-config/pull/847).

**Reply by yan12125** [reply]
> How did it become 1.1.0-2 and not 1.1.0-1?

Arch starts from -1 :)

> only these components should be released:

Yep, thanks!

**Reply by tsujan** [reply]
libqtxdg-3.9.1, qtxdg-tools-3.9.1 and lxqt-session-1.1.1 have been released — hopefully correctly.

**Comment by tenplus1** [comment]
Is their any available repository I can use to test LxQt 1.1 without having to compile it myself ?  Am currently using Lubuntu 22.04.

**Reply by tsujan** [reply]
Lubuntu isn't the best distro for having latest releases without compilation. For that, you need a rolling release (other than Debian, for now).

**Reply by severusseptimius** [reply]
Hi! If you want to test LXQt 1.1 without having to compile it yourself, you can install the packages here: https://github.com/severusseptimius/lxqt-1.1-debian. If you have any questions/problems, you can open an issue there.

**Reply by tsujan** [reply]
That's very good news for Debian.

@severusseptimius, do you have a plan for maintaining LXQt in Debian itself? With Alf's disappearance, Debian's LXQt support disappeared too.

**Reply by severusseptimius** [reply]
I am not sure. Maintaining Debian in LXQt is a much more complex and time consuming task than just making Debian native packages. It is indeed my intention to eventually do this, but I'm not sure it will happen in the very near future.

**Reply by palinek** [reply]
I was also trying to reach debian maintainers offering help to revive LXQt in debian, but didn't get much response. Only @czchen made me a member of LXQt group on salsa.debian.org (I don't know, if I'm allowed to merge PRs and what consequences that could bring), but that's all. I would like to help, but I'm not experienced in debian's workflows, so I would need some menthorship and/or how-tos.

**Comment by nlsmart** [comment]
Hi,
Congratulation to the dev team and contributor.

**Reply by nlsmart** [reply]
LXQt 1.1.0-1 available on Manjaro x86 and arm devices.

**Comment by 0323pin** [comment]
Hi,

I've just finished the packaging of LXQt-1.1.0 on NetBSD and we (me and one more dev) are just testing it for a few days before merging the update.

So far, so good, this release feels really smooth and responsive, as 1.0.0 did.

I've been maintaining LXQt since my initial import of version 0.14.1 and I'm wondering if it would be possible to add the NetBSD logo to https://lxqt-project.org/ ?

![Capture](https://user-images.githubusercontent.com/90570748/164455585-3389b398-fd2a-432d-9f9c-eee85e7e0d19.PNG)

Thanks!

**Reply by stefonarch** [reply]
> I'm wondering if it would be possible to add the NetBSD logo to https://lxqt-project.org/ ?

Sure, will do it! Where can I find a logo to start with?

**Reply by stefonarch** [reply]
never mind, found https://www.netbsd.org/gallery/logos.html

**Reply by 0323pin** [reply]
@stefonarch Sorry for the delay, I was about to post the link.

**Reply by stefonarch** [reply]
https://github.com/lxqt/lxqt.github.io/pull/110
(To tiny characters to be sharp...)

**Reply by 0323pin** [reply]
Maybe you could use just the flag, https://www.netbsd.org/gallery/other-logos.html

**Reply by stefonarch** [reply]
I saw this, but there's no version with transparency, it's already merged, it's not too bad IMHO.

**Comment by 0323pin** [comment]
LXQt-1.1.0 merged on NetBSD

**Reply by stefonarch** [reply]
https://github.com/lxqt/lxqt/wiki/Binary-packages

**Reply by tsujan** [reply]
> https://github.com/lxqt/lxqt/wiki/Binary-packages

What about OpenSUSE Tumbleweed? Is it still with 1.0.0?

**Reply by Chiitoo** [reply]
> https://github.com/lxqt/lxqt/wiki/Binary-packages

Just to nitpick a little, there are no "pre-compiled binary packages" provided by Gentoo... although that might be an optional feature somewhere in the future.  :]

Also, we do have 1.0.0 there, and 1.1.0 should be in very soon as well.

Getting a little off-topic perhaps, but while on that article, I might also drop the 'If you need, do "layman -a qt" and emerge them.' part, because 'layman' is becoming more and more obsolete (should still work), and I think mentioning the qt overlay is enough.  Furthermore, I'm not sure the 'snapshots' is very accurate, since by default those ebuilds will pull 'git master' from upstream (but can be used to pick any commit the user desires).  In Gentoo terms, they're commonly called "live ebuilds".

Thank you all for your work on 1.1.0, and beyond!

**Reply by 0323pin** [reply]
For the sake of correctness, Void has merged 1.1.0 a few days ago.

**Reply by stefonarch** [reply]
My knowledge of Gentoo is very very basic, but not listing it on the "_binary_ packages" wiki is not ok too :)

So Gentoo could have git master or the latest release depending on some commands, right?
 Best would be having ` Gentoo ¹: ....` and 

1)explanation - feel free to write down a paragraph.

>What about OpenSUSE Tumbleweed? Is it still with 1.0.0?

No idea, I just started updating that page with the debian and ubuntu ppa links, while arch and netBSD appereared too. Maybe @jubalh has better knowledge about that.

**Reply by stefonarch** [reply]
> For the sake of correctness, Void has merged 1.1.0 a few days ago.

Thanks, inserted.

**Reply by Chiitoo** [reply]
> My knowledge of Gentoo is very very basic, but not listing it on the "_binary_ packages" wiki is not ok too :)

Yeah, it's fine as is, I just felt like nitpicking on that a little, because normally users compile everything from the upstream sources before installing, though for some bigger builds, such as Firefox, binary versions from upstream are made available via the official Gentoo repository as well.

> So Gentoo could have git master or the latest release depending on some commands, right? Best would be having ` Gentoo ¹: ....` and

I'd list stable releases only in the table.  Generally we recommend people to use those instead of the "live ebuilds" pulling straight git commits, because not every upstream is as stable as LXQt is.  ^^

The Gentoo Packages [1] page for the 'lxqt-meta' package can be used to see the current stable and unstable versions (in Gentoo terms) that are available.  I'd probably only list the highest version in the table; 0.17.0 is there still too, but it will be removed soon since 1.0.0 is stable (in Gentoo terms) already as well.

1. https://packages.gentoo.org/packages/lxqt-base/lxqt-meta

> 1)explanation - feel free to write down a paragraph.

Maybe something like this:

```
LXQt stable releases are made available via the official Gentoo repository.  You can find information about installing [here](https://wiki.gentoo.org/wiki/LXQt).

Additionally, "live ebuilds" using git commits can be found from the Qt overlay.
```

Thanks!

**Reply by stefonarch** [reply]
Done :)

**Comment by sfalken-zz** [comment]
On Tue, 2022-04-26 at 13:25 -0700, Standreas wrote:
> My knowledge of Gentoo is very very basic, but not listing it on the
> "binary packages" wiki is not ok too :)
> So Gentoo could have git master or the latest release depending on
> some commands, right?
> Best would be having  Gentoo ¹: .... and
> 1)explanation - feel free to write down a paragraph.
> > What about OpenSUSE Tumbleweed? Is it still with 1.0.0?
> No idea, I just started updating that page with the debian and ubuntu
> ppa links, while arch and netBSD appereared too. Maybe @jubalh has
> better knowledge about that.
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you are subscribed to this
> thread.Message ID:
> ***@***.***>

As of right now, Tumbleweed is still shipping 1.0.0 from the Main
Repositories.   I am not currently involved in the packaging, but I
would imagine 1.1.0 will be making their way through for approval in
short order.

**Comment by 0323pin** [comment]
Taking 1.1.0 for a spin using a custom build 😄 
https://i.postimg.cc/3N6BCrH4/screen.png

**Comment by peabee** [comment]
> * In LXQt Panel, the legacy tray icons are now shown inside Status Notifier when the System Tray plugin is enabled. That removes the problem that the legacy System Tray had with auto-hiding panels.

Many thanks for the work done in 1.1.0 to support legacy systray items - much appreciated.
This is a screenshot of a package built from ArchLinux LXQt components for a version of Puppy Linux....

![Screenshot](https://user-images.githubusercontent.com/5926865/165985226-534c24c8-882c-4e76-981c-8e14dcd6e3a3.png)

**Comment by jubalh** [comment]
> What about OpenSUSE Tumbleweed? Is it still with 1.0.0?
 No idea, I just started updating that page with the debian and ubuntu
 ppa links, while arch and netBSD appereared too. Maybe [@jubalh](https://github.com/jubalh) has
 better knowledge about that.

I submitted it today. Didn't have time before.

**Comment by stefonarch** [comment]
> Retweet this (or speak up on our other channels) if you'd like to see LXQt 1.1.0 backported to Lubuntu 22.04 LTS
> If it's highly requested, we'll make it happen

https://twitter.com/LubuntuOfficial/status/1531769158073040896
